### PR TITLE
Add guide on error handling (+ Free Applicative)

### LINKED ISSUE #58: Readme and example
We should write a first README with global information about Preface (goal, motivation, roadmap etc.) and having examples. 

Examples should be some concrete use case of Preface (without test boilerplate).

### PR DESCRIPTION
**Pull request by xvw**
I try to give: 
- Sequential error handling using `Result/Try`
- Parallel error handling using `Validation/Validate`
- Canonical representation of Validation pipeline using a `Free Applicative` over `Validate`.
And I guess it Fix #58 

[Rendered version](https://github.com/xvw/preface/blob/guides/error-handling/guides/error_handling.md)

### CODE REVIEW

File: guides/error_handling.md [hunk 1]
```diff
@@ -0,0 +1,756 @@
...
+    return 42 
+       >>= safe_divide 2
+       >|= succ 
+       >|= succ
+       >>= safe_divide 2
+    ;;
+val result : int Preface.Try.Monad.t = Ok 11
+```
+
+We could also perform a calculation that could fail... by dividing by
```
Comment (gr-im) [inline guides/error_handling.md:83]:
```diff
- We could also perform a calculation that could fail... by dividing by
+ We could also perform a computation that could fail... by dividing by
```

Comment (gr-im) [reply guides/error_handling.md:83]: The change is on -calculation +computation

File: guides/error_handling.md [hunk 2]
```diff
@@ -0,0 +1,756 @@
...
+  function
+- We would like to be able to derive a program described in our DSL
+  into an HTML form
+  
+This part of the guide is a fairly loose interpretation of the
+technique described in [The Essence of Form
+Abstraction](https://homepages.inf.ed.ac.uk/slindley/papers/formlets-essence.pdf)
+by E. Cooper, S. Lindley, P. Wadler and J. Yallop.
+
+First, we need something to characterize a form:
```
Comment (gr-im) [inline guides/error_handling.md:317]: `characterise`

**Review by gr-im — CHANGES_REQUESTED** [review]
Hi, it is a very interesting guide. Thanks. Here is some small changement suggestion!

File: guides/error_handling.md [hunk 3]
```diff
@@ -0,0 +1,756 @@
...
+```
+
+Great! Now let's try it with ANYTHING that doesn't follow the rules!
+(And yes, there are plenty of bad users on the net!)
+
+```ocaml
+# create_user "x" (-23) "abademail" ;;
+- : user Preface_stdlib.Validate.Selective.t =
+Preface_stdlib__.Validation.Invalid
+ [Nickname_to_short "x"  Invalid_age (-23)  Invalid_email "abademail"]
```
[the remark this answers is not in the document]
Comment (xvw) [reply guides/error_handling.md]: Yes, it was a mstake (and now it is fixed for Nonempty_list and List ;) nice spot)

File: guides/error_handling.md [hunk 4]
```diff
@@ -0,0 +1,756 @@
...
+**validated users**:
+
+```ocaml
+module Validated_list = 
+   Preface.List.Applicative.Traversable 
+       (Preface.Validate.Applicative)
+```
+
+Now we can use, for example, the `sequence` function whose type is:
+`'a Validate.t list -> 'a list Validate,t` which seems to be exactly
```
[the remark this answers is not in the document]
Comment (xvw) [reply guides/error_handling.md]: EagleEye

**Review by gr-im — APPROVED** [review]
Thanks for the changes and the fixes.

**Review by d-plaindoux — CHANGES_REQUESTED** [review]

File: guides/error_handling.md [hunk 5]
```diff
@@ -0,0 +1,756 @@
...
+
+- [Result](https://ocaml-preface.github.io/preface/Preface_stdlib/Result/index.html)
+  `Result` is a type that describes **sequential error validation** quite
+  well. In a pipeline of failable computing, as soon as a function
+  fails, the calculation is interrupted.
+- [Either](https://ocaml-preface.github.io/preface/Preface_stdlib/Either/index.html)
+  `Either` is strictly equivalent to `Result` (`('a, 'b) Either.t =
+  ('b, 'a) Result.t`). But its constructors carry less meaning, in the
+  context of error handling, so we will not use them in this guide.
+- [Try](https://ocaml-preface.github.io/preface/Preface_stdlib/Try/index.html)
```
[the remark this answers is not in the document]
Comment (xvw) [reply guides/error_handling.md:21]: I put `Either` after the enumeration for more clarity, is that oké for you?

File: guides/error_handling.md [hunk 6]
```diff
@@ -0,0 +1,756 @@
...
+  As `Result` is parameterized by two types, `Try` is a biased version
```
[the remark this answers is not in the document]
Comment (xvw) [reply guides/error_handling.md]: sure, it is done.

File: guides/error_handling.md [hunk 7]
```diff
@@ -0,0 +1,756 @@
...
+- [Validate](https://ocaml-preface.github.io/preface/Preface_stdlib/Validate/index.html)
+  As `Validation` is parameterized by two types, `Validate` is a
+  biased version of `Validation`. Its error type is set to be a
+  non-empty list of exception, because if there is an error, then at
+  least one error has occurred. So it wouldn't make sense to have a
+  normal list of errors. (`'a Validate.t = ('a, exn Nonempty_list.t)
+  Validation.t`).
+
+As exceptions are an open sum type, they are comfortable for declaring
+errors on the fly. However, to ensure completeness of error handling,
```
[the remark this answers is not in the document]
Comment (xvw) [reply guides/error_handling.md]: done.

File: guides/error_handling.md [hunk 8]
```diff
@@ -0,0 +1,756 @@
...
+provided that the validation is sequential. There is an extremely
+pedagogical presentation by Scott Wlaschin on this approach, [Railway
+oriented programming: Error handling in functional
+languages](https://vimeo.com/113707214), and I suggest you watch it if
+you want to see more scenarios.
+
+
+## Parallel validation
+
+But sometimes sequential validation can be an anti-pattern. For
```
[the remark this answers is not in the document]
Comment (xvw) [reply guides/error_handling.md]: I change it for dark-pattern. The sentence is explicit that can be "sometime" a dark pattern.

**Review by d-plaindoux — APPROVED** [review]

**Pull request merged**